docs/app/docs/ide_configuration/direnv.md

```diff
@@ -0,0 +1,43 @@
+---
+title: direnv configuration
+---
+
+
+## direnv
+___
+[direnv](https://direnv.net) is an open source environment management tool that allows setting unique environment variables per directory in your file system. This guide covers how to configure direnv to seemlessly work with a devbox project.
+
+### Prerequisites
+* Install direnv and hook it to your shell. Follow [this guide](https://direnv.net/#basic-installation) if you haven't done it. 
+
+### Setting up with Devbox Shell and direnv
+
+Note: If you already have a devbox project you may skip to step 3.
+
+1. `devbox init` if you don't have a devbox.json in the root directory of your project.
+2. `devbox shell -- 'ls'` to activate devbox shell temporarily and make sure dependencies mentioned in your devbox.json are installed.
+3. Create a new file, name it `.envrc` and put the following snippet inside it:
+    ```bash
+    use_devbox() {
+        watch_file devbox.json
+        eval $(devbox setup-direnv)
+    }
+    use devbox
+    ```
+4. Run `direnv allow` to give permission to `direnv` to setup your environment variables.
+5. At this point your project directory is setup so that everytime you `cd` into it, the binaries from your devbox shell will be used. To test this, you can compare running `which python3` from your project directory and outside.
+
+### Global settings for direnv
+
+Note that everytime changes are made to devbox.json via `devbox add ...`, `debox rm ...` or direct editing of the file, `direnv allow` will be required to run so that `direnv` can setup the new changes. 
```

Comment:
nits:
- backticks around devbox.json
- everytime -> every time
- debox rm -> devbox rm
- direct editing of the file -> directly editing the file